drivers/sensor/mhz19b/mhz19b.c

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2021 G-Technologies Sdn. Bhd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.winsen-sensor.com/sensors/co2-sensor/mh-z19b.html
+ */
+
+#define DT_DRV_COMPAT winsen_mhz19b
+
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+
+#include <drivers/sensor/mhz19b.h>
+#include "mhz19b.h"
+
+LOG_MODULE_REGISTER(mhz19b, CONFIG_SENSOR_LOG_LEVEL);
+
+static void mhz19b_uart_flush(const struct device *uart_dev)
+{
+	uint8_t c;
+
+	while (uart_fifo_read(uart_dev, &c, 1) > 0) {
+		continue;
+	}
+}
+
+static uint8_t mhz19b_checksum(const uint8_t *data)
+{
+	uint8_t cs = 0;
+
+	for (uint8_t i = 1; i < MHZ19B_BUF_LEN - 1; i++) {
+		cs += data[i];
+	}
+
+	return 0xff - cs + 1;
+}
+
+static int mhz19b_send_cmd(const struct device *dev, const enum mhz19b_cmd_idx cmd_idx,
+			   const bool has_rsp)
```

Comment:
what is the value of adding `const` here (and others)?

Reply:
Do you mean all three `const`s in the args?

Reply:
I added `const` to all of them because they are constants in this function

Reply:
I've removed all the `const` for non-ptr type args